### PR TITLE
fix: Make MySQL migration for nano id change compatible with version 5.7

### PR DESCRIPTION
**Pull request by krynble**
Github issue / Community forum post (link here to close automatically):

### CODE REVIEW

**Review by flipswitchingmonkey — DISMISSED** [review]

**Review by netroy — COMMENTED** [review]
[Started DB tests](https://github.com/n8n-io/n8n/actions/runs/5332571860/jobs/9661927713). Let's wait for them to pass, before merging.

**Review by netroy — DISMISSED** [review]
[Tests failed](https://github.com/n8n-io/n8n/actions/runs/5332571860/jobs/9661927713)

File: packages/cli/src/databases/migrations/mysqldb/1690000000001-MigrateIntegerKeysToString.ts [hunk 1]
```diff
@@ -1,38 +1,55 @@
...
+const COLLATION_80 = 'utf8mb4_0900_ai_ci';
+
 export class MigrateIntegerKeysToString1690000000001 implements ReversibleMigration {
 	async up({ queryRunner, tablePrefix }: MigrationContext) {
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+		const dbVersionQuery = (await queryRunner.query('SELECT @@version')) as  // eslint-disable-next-line @typescript-eslint/naming-convention
+			| Array<{ '@@version': string }>
+			| undefined;
+		let collation = COLLATION_80;
+		if (dbVersionQuery?.length === 1) {
```
Comment (netroy) [inline packages/cli/src/databases/migrations/mysqldb/1690000000001-MigrateIntegerKeysToString.ts:13]: [we have a similar check in a previous migration](https://github.com/n8n-io/n8n/blob/master/packages/cli/src/databases/migrations/mysqldb/1623936588000-CertifyCorrectCollation.ts#L10). maybe we can unify the two, or at least use the same code in both the places.

File: packages/cli/src/databases/migrations/mysqldb/1690000000001-MigrateIntegerKeysToString.ts [hunk 2]
```diff
@@ -1,38 +1,55 @@
 import type { MigrationContext, ReversibleMigration } from '@db/types';
 
+const COLLATION_57 = 'utf8mb4_general_ci';
+const COLLATION_80 = 'utf8mb4_0900_ai_ci';
+
 export class MigrateIntegerKeysToString1690000000001 implements ReversibleMigration {
 	async up({ queryRunner, tablePrefix }: MigrationContext) {
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+		const dbVersionQuery = (await queryRunner.query('SELECT @@version')) as  // eslint-disable-next-line @typescript-eslint/naming-convention
```
Comment (netroy) [inline packages/cli/src/databases/migrations/mysqldb/1690000000001-MigrateIntegerKeysToString.ts:9]: I think we might need skip the whole collation thing on mariadb.

**Review by flipswitchingmonkey — APPROVED** [review]

**Review by netroy — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@0.234.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%400.234.0)

**Pull request merged**